examples/mdetr/utils/misc.py

```diff
@@ -0,0 +1,126 @@
+    if "positive_map_eval" in batch[1][0]:
+        # we batch the positive maps here
+        # Since in general each batch element will have a different number of boxes,
+        # we collapse a single batch dimension to avoid padding. This is sufficient for our purposes.
+        max_len = max([v["positive_map_eval"].shape[1] for v in batch[1]])
+        nb_boxes = sum([v["positive_map_eval"].shape[0] for v in batch[1]])
+        batched_pos_map = torch.zeros((nb_boxes, max_len), dtype=torch.bool)
+        cur_count = 0
+        for v in batch[1]:
+            cur_pos = v["positive_map_eval"]
+            batched_pos_map[
+                cur_count : cur_count + len(cur_pos), : cur_pos.shape[1]
+            ] = cur_pos
+            cur_count += len(cur_pos)
+
+        assert cur_count == len(batched_pos_map)
+        final_batch["positive_map_eval"] = batched_pos_map.float()
+    if "answer_type_mask" in batch[1][0]:
+        answer_types = {
+            k: torch.cat([b["answer_type_mask"][k] for b in batch[1]])
+            for k in batch[1][0]["answer_type_mask"].keys()
+        }
+        final_batch["answer_type_mask"] = answer_types
+
+    if "answer" in batch[1][0]:
+        answers = {}
+        for f in batch[1][0].keys():
+            if (
+                "answer" not in f or f == "answer" or f == "answer_type_mask"
+            ):  # We only use split_qa_heads = True
+                continue
+            answers[f] = torch.stack([b[f] for b in batch[1]])
+        final_batch["answers"] = answers
+    final_batch["batch_encoding"] = tokenizer.batch_encode_plus(
+        [v["caption"] for v in batch[1]], padding="longest", return_tensors="pt"
+    ).to(batched_pos_map.device)
+    return final_batch
+
+
+def interpolate(
+    input: Tensor,
+    size: Optional[List[int]] = None,
+    scale_factor: Optional[float] = None,
+    mode: str = "nearest",
+    align_corners: Optional[bool] = None,
+):
+    # type: (Tensor, Optional[List[int]], Optional[float], str, Optional[bool]) -> Tensor
+    """
+    Equivalent to nn.functional.interpolate, but with support for empty channel sizes.
+    """
+    if input.numel() > 0:
+        return torch.nn.functional.interpolate(
+            input, size, scale_factor, mode, align_corners
+        )
+
+    assert (
+        input.shape[0] != 0 or input.shape[1] != 0
+    ), "At least one of the two first dimensions must be non zero"
+
+    if input.shape[1] == 0:
+        # Pytorch doesn't support null dimension on the channel dimension, so we transpose to fake a null batch dim
+        return torch.nn.functional.interpolate(
+            input.transpose(0, 1), size, scale_factor, mode, align_corners
+        ).transpose(0, 1)
+
+    # empty batch dimension is now supported in pytorch
+    return torch.nn.functional.interpolate(
+        input, size, scale_factor, mode, align_corners
+    )
+
+
+def targets_to(targets: List[Dict[str, Any]], device):
```

Comment:
used where?

Reply:
Used later in the stack (e.g. phrase_grounding.py in #181 )